src/__swaps__/screens/Swap/components/TokenList/ChainSelection.tsx

```diff
@@ -0,0 +1,248 @@
+/* eslint-disable no-nested-ternary */
+/* eslint-disable react/jsx-props-no-spreading */
+import c from 'chroma-js';
+import { Text as RNText, StyleSheet } from 'react-native';
+import Animated, { SharedValue, runOnUI, useAnimatedReaction, useSharedValue } from 'react-native-reanimated';
+import React, { useCallback, useMemo } from 'react';
+
+import { SUPPORTED_CHAINS } from '@/references';
+import { AnimatedText, Bleed, Box, HitSlop, Inline, Text, globalColors, useColorMode, useForegroundColor } from '@/design-system';
+import { chainNameFromChainId, chainNameFromChainIdWorklet } from '@/__swaps__/utils/chains';
+import { opacity } from '@/__swaps__/utils/swaps';
+import { ethereumUtils, showActionSheetWithOptions } from '@/utils';
+import { ChainImage } from '@/components/coin-icon/ChainImage';
+import { ChainId, ChainName } from '@/__swaps__/types/chains';
+import { useSwapContext } from '@/__swaps__/screens/Swap/providers/swap-provider';
+import { ContextMenuButton } from '@/components/context-menu';
+import { useAccountAccentColor } from '@/hooks';
+import { UserAssetFilter } from '@/__swaps__/types/assets';
+import { OnPressMenuItemEventObject } from 'react-native-ios-context-menu';
+import { Network } from '@/helpers';
+
+type ChainSelectionProps = {
+  allText?: string;
+  output: boolean;
+};
+
+export const ChainSelection = ({ allText, output }: ChainSelectionProps) => {
+  const { isDarkMode } = useColorMode();
+  const { accentColor: accountColor } = useAccountAccentColor();
+  const { SwapInputController, userAssetFilter } = useSwapContext();
+  const red = useForegroundColor('red');
+
+  const accentColor = useMemo(() => {
+    if (c.contrast(accountColor, isDarkMode ? '#191A1C' : globalColors.white100) < (isDarkMode ? 2.125 : 1.5)) {
+      const shiftedColor = isDarkMode ? c(accountColor).brighten(1).saturate(0.5).css() : c(accountColor).darken(0.5).saturate(0.5).css();
+      return shiftedColor;
+    } else {
+      return accountColor;
+    }
+  }, [accountColor, isDarkMode]);
+
+  const propToSet = output ? SwapInputController.outputChainId : userAssetFilter;
+
+  const chainName = useSharedValue(
+    propToSet.value === 'all' ? allText : propToSet.value === ChainId.mainnet ? 'ethereum' : chainNameFromChainIdWorklet(propToSet.value)
+  );
+
+  useAnimatedReaction(
+    () => ({
+      outputChainId: SwapInputController.outputChainId.value,
+      userAssetFilter: userAssetFilter.value,
+    }),
+    current => {
+      if (output) {
+        chainName.value = current.outputChainId === ChainId.mainnet ? 'ethereum' : chainNameFromChainIdWorklet(current.outputChainId);
```

Comment:
looks like this is happening in a bunch of places, may be worth just addint the mainnet -> ethereum mapping inside of the worklet